tests/v2/validation/provisioning/rke2/rke2_encryption_key_rotation_test.go

```diff
@@ -100,35 +104,63 @@ func (r *RKE2EncryptionKeyRotationTestSuite) TestEncryptionKeyRotationImpl(provi
 		assert.NoError(r.T(), err)
 		assert.Equal(r.T(), clusterName, clusterResp.Name)
 
-		provCluster, err := r.client.Provisioning.Clusters(namespace).Get(context.TODO(), clusterName, metav1.GetOptions{})
-		assert.NoError(r.T(), err)
-		if err != nil {
-			return
-		}
+		//require.NoError(r.T(), r.rotateEncryptionKeys(clusterName, 1))
+		//logrus.Infof("Successfully completed encryption key rotation for %s", name)
 
-		provCluster = provCluster.DeepCopy()
-		provCluster.Spec.RKEConfig.RotateEncryptionKeys = &rkev1.RotateEncryptionKeys{
-			Generation: 1,
+		// Get clusterID with clusterName
+		clusterID, err := clusters.GetClusterIDByName(r.client, clusterName)
+		require.NoError(r.T(), err)
+
+		// create 1000 secrets
+		for i := 0; i < 1000; i++ {
+			s := &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					GenerateName: fmt.Sprintf("encryption-key-rotation-test-%d-", i),
+				},
+				Data: map[string][]byte{
+					"key": []byte(namegenerator.RandStringLower(5)),
+				},
+			}
+			_, err = secrets.CreateSecret(r.client, s, clusterName, "default")
```

Comment:
@igomez06 Any idea why this sin't working? It always errors with "the server could not find the requested resource"

Reply:
It's because you are passing the cluster name rather than the clusterID

Reply:
Oh, duh